lib/ansible/modules/cloud/memset/memset_memstore_user.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Simon Weald <ansible@simonweald.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: memset_memstore_user
+author: "Simon Weald (@glitchcrab)"
+version_added: "2.9"
+short_description: Manage Memstore users
+notes:
+  - Create/delete and manage the status of Memstore users.
+  - An API key generated via the Memset customer control panel is needed with
+    the following minimum scope - I(memset.user_list), I(memstore.user_info),
+    I(memstore.user_enable), I(memstore.user_disable), I(memstore.user_delete),
+    I(memstore.user_create).
+description:
+    - Create/delete and manage the status of Memstore users.
+options:
+    state:
+        required: true
+        default: present
+        description:
+            - Indicates desired state of resource.
+        choices: [ absent, present ]
+    api_key:
+        required: true
+        description:
+            - The API key obtained from the Memset control panel.
+    username:
+        required: true
+        description:
+            - The name of the user to manage.
+        aliases: [ name ]
+    memstore:
+        required: true
+        description:
+            - The Memstore product the user belongs to.
+    enabled:
+        required: false
+        default: false
+        type: bool
+        description:
+            - Whether the user is enabled or disabled. Defaults to false for security.
+    password:
+        required: false
+        description:
+            - A password for the user. Required when the user is present.
+            - I(update_password) must be True for an existing user's password to be updated.
+    update_password:
```

Comment:
This should behave the same as the identically named option on the `user` module.

Reply:
sorry, would you mind explaining what you mean by this?

Reply:
See https://docs.ansible.com/ansible/latest/modules/user_module.html There, `update_password` is a string, which can be `always` (default) or `on_create`. The first means that the password will always be set, and the second that it will only be set on user creation, but not changed later on.

Reply:
Right, so having taken a look at this it's going to be a little difficult to replicate the same behaviour for this module. For reasons decided before my time, it's impossible to tell if the password has been changed so therefore I can't replicate the behaviour of `always`. The API will return changed every time you submit a password update, even if it's the same password as already set. 

What do you suggest I do here?

Reply:
You could add a comment to the docs that a value of "always" will always result in a changed result because the API cannot tell whether the password was changed. @jamescassell what do you think?